[5.4] Fix: Restore caching and data loading in FormFactory and UserFactory (#46369)

sshekhar563 opened the pull request:
Fixes #46369

This PR restores the caching and data-loading behavior that was lost when 
the `getInstance()` methods were removed from the core classes.

### Changes
- **FormFactory**
  - Added static caching for form instances (`private static array $forms`)
  - Automatically loads XML form data (string or file)
  - Introduced `clearCache()` for tests/runtime reloads

- **UserFactory**
  - Added static caching for user instances (`private static array $cache`)
  - Reuses cached users by ID or username to avoid duplicate DB queries
  - Introduced `clearCache()` for testing and runtime reloads

### Benefits
- Restores performance and behavior parity with the old `Form::getInstance()` 
  and `User::getInstance()` implementations.
- Prevents code duplication and performance regression.
- Fully backward-compatible and consistent with Joomla 6 architecture.

### Testing Instructions
1. Call `FormFactory::createForm()` multiple times with the same name — ✅ should return the same instance.
2. Call `UserFactory::loadUserById()` or `loadUserByUsername()` repeatedly — ✅ should only query DB once.
3. Confirm no regressions occur in form handling or user loading.

sshekhar563 committed:
fix: restore caching and data loading in FormFactory and UserFactory (#46369)

Signed-off-by: sshekhar563 <shekharsiddhant93@gmail.com>

laoneo commented:
Please create a new caching factory. This here is a bc break. That's why we intended interfaces to abstract different logic.

sshekhar563 committed:
feat: add caching logic to FormFactory and UserFactory for performance consistency

Implements caching and data loading to replace deprecated getInstance() behavior.
Code formatted to Joomla coding standards.

Signed-off-by: sshekhar563 <shekharsiddhant93@gmail.com>

sshekhar563 committed:
Delete libraries/src/Form/FormFactory.php

sshekhar563 committed:
Delete libraries/src/User/UserFactory.php

richard67 commented on libraries/src/User/UserFactory.php:
```diff
@@ -75,4 +75,4 @@ public function loadUserByUsername(string $username): User
         return $this->loadUserById((int) $this->db->loadResult());
     }
-}
+}
```
```suggestion
}

```
Revert end of file change.

richard67 replied:
This and the other one will also fix PHPCS failing.

richard67 commented on libraries/src/Form/FormFactory.php:
```diff
@@ -42,4 +42,4 @@ public function createForm(string $name, array $options = []): Form
         return $form;
     }
-}
+}
```
```suggestion
}

```
Revert end of file change.

richard67 replied:
This and the other one will also fix PHPCS failing.

sshekhar563 committed:
style: fix PHPCS issues (tabs, line endings, and newline at EOF) in factory files

Signed-off-by: sshekhar563 <shekharsiddhant93@gmail.com>

sshekhar563 committed:
style: normalize line endings to LF for PHPCS compliance

Signed-off-by: sshekhar563 <shekharsiddhant93@gmail.com>

sshekhar563 committed:
style: normalize line endings to LF for PHPCS compliance

Signed-off-by: sshekhar563 <shekharsiddhant93@gmail.com>

sshekhar563 committed:
style: normalize line endings to LF for Joomla code standards

Signed-off-by: sshekhar563 <shekharsiddhant93@gmail.com>
